tests/pages/api/event/[id]/repair-request/get-filtered.test.ts

```diff
@@ -0,0 +1,196 @@
+      description: "Laptop repair event.",
+      startDate: new Date("2023-11-16"),
+      endDate: new Date("2023-11-17")
+    };
+    const eventIds = ["ev-1", "ev-2"];
+    await prisma.event.createMany({
+      data: eventIds.map((id) => ({ ...baseEventDetails, id, name: id }))
+    });
+
+    // Create repair requests
+    const baseRepairRequestDetails = {
+      comment: "Some comment",
+      thumbnailImage: ""
+    };
+    await prisma.repairRequest.createMany({
+      data: [
+        {
+          ...baseRepairRequestDetails,
+          id: "rr-1",
+          eventId: "ev-1",
+          createdBy: "user_1",
+          description: "My laptop is broken",
+          itemType: "Laptop",
+          itemBrand: "Apple",
+          assignedTo: "user_2"
+        },
+        {
+          ...baseRepairRequestDetails,
+          id: "rr-2",
+          eventId: "ev-1",
+          createdBy: "user_1",
+          description: "Very specific word: eggplant",
+          itemType: "Clock",
+          itemBrand: "Clock Co"
+        },
+        {
+          ...baseRepairRequestDetails,
+          id: "rr-3",
+          eventId: "ev-2",
+          createdBy: "user_2",
+          description: "My laptop is broken",
+          itemType: "Laptop",
+          itemBrand: "Dell"
+        },
+        {
+          ...baseRepairRequestDetails,
+          id: "rr-4",
+          eventId: "ev-2",
+          createdBy: "user_1",
+          description: "Yep",
+          itemType: "Laptop",
+          itemBrand: "Dell"
+        }
+      ]
+    });
+  });
+  type AllowedParams = {
+    id?: string;
+    sortKey?: string;
+    sortMethod?: string;
+    searchWord?: string;
+    itemType?: string | string[];
+    brand?: string | string[];
+    assignedTo?: string | string[];
+    page?: string;
+    perPage?: string;
+  };
+  const testFilter = async (
+    filters: AllowedParams,
+    expectedEvents: string[]
+  ) => {
+    await testApiHandler({
+      handler,
+      params: { ...filters },
+      test: async ({ fetch }) => {
+        const res = await fetch({
+          method: "GET",
+          headers: {
+            "Content-Type": "application/json"
+          }
+        });
+        const results: PaginationResponse<RepairRequestResponse> =
+          await res.json();
+        expect(res.status).toBe(200);
+        expect(results.items.length).toBe(expectedEvents.length);
+        results.items.forEach((result, index) => {
+          expect(result).toHaveProperty("id", expectedEvents[index]);
+        });
+      }
+    });
+  };
+  const testBadFilter = async (filters: AllowedParams, code: number = 400) => {
+    await testApiHandler({
+      handler,
+      params: { ...filters },
+      test: async ({ fetch }) => {
+        const res = await fetch({
+          method: "GET",
+          headers: {
+            "Content-Type": "application/json"
+          }
+        });
+        expect(res.status).toBe(code);
+      }
+    });
+  };
```

Comment:
could some of these be moved to separate files in the same directory eg. utils, mock-data, etc? It makes this file easier to digest and read.